backend/dao/winesDAO.js

```diff
@@ -88,85 +63,67 @@ function veganSelectors(filters) {
 //* *******************************
 //* *******************************
 
-function profileSelectors(filters) {
-  if (filters.profile.includes("dry") && filters.profile.includes("sweet")) {
-    do {
-      for (var i = filters.profile.length - 1; i >= 0; i--) {
-        if (filters.profile[i] === "sweet") {
-          filters.profile.splice(i, 1);
-        }
-      }
-      for (var i = filters.profile.length - 1; i >= 0; i--) {
-        if (filters.profile[i] === "dry") {
-          filters.profile.splice(i, 1);
-        }
-      }
-    } while (
-      filters.profile.includes("dry") ||
-      filters.profile.includes("sweet")
+function profileSelectors(profileFilter) {
+  let newProfileFilter = [];
+  // if just one filter is passed, profileFilter will be a string
+  if (typeof profileFilter === "string") {
+    newProfileFilter = [profileFilter];
+  } else {
+    newProfileFilter = [...profileFilter];
+  }

```

Comment:
This is a repeated, it could be extracted to a separate function, but not that much mandatory because there are just 2 repetitions for now.